### ISSUE TITLE
Make length.(keys) work

### ISSUE THREAD

**Issue by cossio** [JuliaLang/julia#19577]
Try this:

```
d = Dict([1,2] => 1.1, [3,2] => 0.1)
length.(keys(d))
```

It doesn't work on 0.5 (I've been told it works on master, https://discourse.julialang.org/t/length-keys-d-does-not-work/913). Can we pull this into 0.5.1?

**Comment by nalimilan** [comment]
Looks like I was mistaken. Here's what this returns on master:
```julia
julia> length.(keys(d))
2
```

So actually it just falls back to `length(keys(d))`. That's a more general problem with iterators, as `length.(1:2 for i in 1:3)` returns `3` rather than `[2, 2, 2]` too. I guess that's a matter of defining `broadcast_indices` for these objects.

Ironically, it's easier to fix on 0.5 than on master, since defining `Base.size(x::Base.KeyIterator) = (length(x),)` is enough for the former, but not for the latter.

**Comment by mbauman** [comment]
Mostly a duplicate of #20678/10092.
